[ADD] 예약 생성 기능 구현 (ADD view for appointment creation function)

jhwang0801 opened the pull request:
## :: 최근 작업 주제
- [x] 기능 추가
- [ ] 리뷰 반영
- [ ] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 
### - 예약 생성 기능 구현
    - Form-Data 형식으로 필요한 값을 전달받아 DB에 저장

<br />

## :: 테스트 코드 결과
<img width="578" alt="image" src="https://user-images.githubusercontent.com/75832544/177151472-c90634e8-9a0b-4532-9e1f-0089b30b1f31.png">

jhwang0801 committed:
ADD view for appointment creation function

- add test code
- add limitation to not allow to upload images more than six
- add transaction atomic to prevent different transactions from interfering with one another
- add exceptions

jhwang0801 committed:
ADD view for appointment creation function

- add test code
- add limitation to not allow to upload images more than six
- add transaction atomic to prevent different transactions from interfering with one another
- add exceptions

jhwang0801 committed:
Fix code convention in Appointment creation testcode

jasonkang14 commented on appointments/views.py:
```diff
@@ -96,5 +97,52 @@ def patch(self, request, appointment_id):
+            if len(images) > 6:
+                return JsonResponse({'message' : 'DO_NOT_ALLOW_TO_UPLOAD_IMAGES_MORE_THAN_6'}, status=400)
+
+            with transaction.atomic():
```
💯

jasonkang14 commented on appointments/views.py:
```diff
@@ -96,5 +97,52 @@ def patch(self, request, appointment_id):
+                return JsonResponse({'message' : 'DO_NOT_ALLOW_TO_UPLOAD_IMAGES_MORE_THAN_6'}, status=400)
+
+            with transaction.atomic():
+                Appointment.objects.create(
```
여기서 선언을 
```python
new_appointment = Appointment.objects.create(
  ...
)
```
이런식으로 해주시면
아래에서`get()` 을 사용한 DB접근을 안해도 됩니다
```python
 UserAppointment.objects.create(
                    appointment_id = new_appointment.id,
                    doctor_id      = doctor_id,
                    patient_id     = patient_id
                )

```

jasonkang14 replied:
그리고 아래와 같은 경우에는 혹시나 같은날 같은 시간에 예약이 존재한다면 (다른 의사, 다른 환자 등) 에러가 발생할 수 있을 것 같습니다

jhwang0801 replied:
아!!! 위에서 선언을 해주는 방법을 생각 못했습니다.

말씀해주신대로 다른 유저가 같은 의사, 같은 날, 같은 시간에 예약이 우연하게 되는 경우에 대비하기 위해서 `Appointment.MultipleObjectsReturned` 예외처리를 해주었는데,

애초에 선언해서 받으면 해당 에러가 발생할 가능성 자체가 없겠네요!!

해당 부분 수정하도록 하겠습니다! 감사합니다!!

jhwang0801 committed:
Update view to solve the possible error

- remove testcode regarding to MultipleObjectsReturned
- remove exceptions regarding to MultipleObjectsReturned and DoesNotExist

jhwang0801 committed:
Fix testcode in AppointmentCreationTest

jasonkang14 reviewed (approved):
LGTM
